nomic/embed.py

```diff
@@ -115,75 +143,46 @@ def images(images: Union[str, PIL.Image.Image], model: str = 'nomic-embed-vision
     Returns:
         An object containing your embeddings and request metadata
     """
+    global atlas_class
 
-    def run_inference(batch):
-        response = requests.post(
-            atlas_class.atlas_api_path + "/v1/embedding/image",
-            headers=atlas_class.header,
-            data={"model": "nomic-embed-vision-v1"},
-            files=batch,
-        )
+    if atlas_class is None:
+        atlas_class = AtlasClass()
 
-        if response.status_code == 200:
-            return response.json()
-        else:
-            print(response.text)
-            raise Exception(response.status_code)
-
-    def resize_pil(img):
-        width, height = img.size
-        #if image is too large, downsample before sending over the wire
-        max_width = 512
-        max_height = 512
-        if max_width > 512 or max_height > 512:
-            downsize_factor = max(width/max_width, height/max_height)
-            img.resize((width/downsize_factor, height/downsize_factor))
-        return img
-
-    def send_request(i):
-        image_batch = []
-        shard = images[i:i+IMAGE_EMBEDDING_BATCH_SIZE]
-        # process images into base64 encoded strings (for now)
-        for image in shard:
-            # TODO implement check for bytes.
-            # TODO implement check for a valid image.
-            if isinstance(image, str) and os.path.exists(image):
+    max_workers = 10
+    chunksize = MAX_IMAGE_REQUEST_SIZE
+    smallchunk = max(1, int(len(images) / max_workers))
+    # if there are fewer texts per worker than the max chunksize just split them evenly
+    chunksize = min(smallchunk, chunksize)
+
+    image_batch = []
+    for image in images:
+        if isinstance(image, str) and os.path.exists(image):
                 img = PIL.Image.open(image)
```

Comment:
as-is we wouldn't resize images passed as file names - prolly want to wrap this in resize_pil to be consistent